File: apps/monolith/src/main/java/com/example/iter/admin/action/util/AdminActionMapper.java
package com.example.iter.admin.action.util;

import com.example.iter.common.audit.domain.entity.AdminAction;
import com.example.iter.admin.action.dto.response.AdminActionResponse;
import org.springframework.stereotype.Component;

@Component
public class AdminActionMapper {

    // 관리자 처리 이력 엔티티를 처리 이력 응답으로 변환합니다.
    public AdminActionResponse toResponse(AdminAction adminAction) {
        return new AdminActionResponse(
                adminAction.getId(),
                adminAction.getAdminId(),
                adminAction.getTargetType(),
                adminAction.getTargetId(),
                adminAction.getAction(),
                adminAction.getReason(),
                adminAction.getCreatedAt()
        );
    }
}

File: apps/monolith/src/main/kotlin/com/example/iter/admin/action/util/AdminActionMapper.kt
package com.example.iter.admin.action.util

import com.example.iter.admin.action.dto.response.AdminActionResponse
import com.example.iter.common.audit.domain.entity.AdminAction
import org.springframework.stereotype.Component

@Component
class AdminActionMapper {
    fun toResponse(adminAction: AdminAction): AdminActionResponse = AdminActionResponse(
        adminAction.id,
        adminAction.adminId,
        adminAction.targetType,
        adminAction.targetId,
        adminAction.action,
        adminAction.reason,
        adminAction.createdAt
    )
}
